### PR TITLE
[SCRIPT] conversion script for BERT

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
- Added script to compare gluon and tf model inference result
- Added script to convert tf model checkpoint to gluon parameters.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scripts/bert/utils.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Utility functions for BERT."""
+import json
+import hashlib
+import gluonnlp
+from tensorflow.python import pywrap_tensorflow
```
Comment (sxjscience) [inline scripts/bert/utils.py]: This forces the users to install tf in order to run all the Bert experiments in Gluon.

Comment (eric-haibin-lin) [reply scripts/bert/utils.py]: This script is currently only used by convert.py. Other scripts will not import this file. Maybe I should rename the file to make it clearer.

File: scripts/bert/compare_tf_gluon_model.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        layers['values'] = layer_output
+        all_layers.append(layers)
+    tensorflow_out_features = collections.OrderedDict()
+    tensorflow_out_features['layers'] = all_layers
+    tensorflow_all_out_features.append(tensorflow_out_features)
+
+    output_json['features'] = tensorflow_all_out_features
+    tensorflow_all_out.append(output_json)
+
+tf_outputs = (tensorflow_all_out[0]['features'][0]['layers'][t]['values'] for t in layer_indexes)
```
Comment (fierceX) [inline scripts/bert/compare_tf_gluon_model.py]:
```suggestion
tf_outputs = [tensorflow_all_out[0]['features'][0]['layers'][t]['values'] for t in layer_indexes)]
```

**Comment by mli** [comment]
Job PR-449/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-449/6/index.html

**Comment by mli** [comment]
Job PR-449/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-449/7/index.html

**Comment by mli** [comment]
Job PR-449/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-449/8/index.html

File: docs/examples/sentence_embedding/bert.md [hunk 3]
```diff
@@ -157,7 +157,7 @@ generate valid length
 
 ```{.python .input}
 # use the vocabulary from pre-trained model for tokenization
-tokenizer = tokenization.FullTokenizer(vocabulary, do_lower_case=True)
+tokenizer = tokenizer.FullTokenizer(vocabulary, do_lower_case=True)
```
Comment (szha) [inline docs/examples/sentence_embedding/bert.md]: This would overwrite the module variable ‘tokenizer’

**Comment by mli** [comment]
Job PR-449/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-449/10/index.html

**Pull request merged**